pico/Operators/FlatMap.hpp

```diff
@@ -93,18 +81,91 @@ class FlatMap: public UnaryOperator<In, Out> {
 		return new impl_t(parallelism, flatmapf);
 	}
 
+	ff::ff_node *opt_node(int par, PEGOptimization_t opt, StructureType st, //
+			opt_args_t a) {
+		assert(false);
+	}
+
+	std::function<void(In&, FlatMapCollector<Out> &)> flatmapf;
+
+};
+
+/**
+ * Defines an operator performing a FlatMap, taking in input one element from
+ * the input source and producing zero, one or more elements in output.
+ * The FlatMap kernel is defined by the user and can be a lambda function, a functor or a function.
+ *
+ * It implements a data-parallel operator that ignores any kind of grouping or windowing.
+ *
+ * The kernel is applied independently to all the elements of the collection (either bounded or unbounded).
+ */
```

Comment:
I would move the comment above to the beginning of the file, rephrasing as "this file defines operators performing a FlatMap..." and I would add a 1-line comment on top of each class, e.g., "this is the base class for the FlatMap operators" etc.